Since b6676bf cmake_modules cannot be found

Packages can't find  `cmake-modules` anymore. On [dde852a](https://github.com/lopsided98/nix-ros-overlay/commit/dde852afc9f761d150950d44c8b03ee5a19dc73e)

Bisected to b6676bf7dc1b03afe9582c5fd99025b3a4b7f4f9

```console
[jkolb@nixos:~/nix-ros-overlay]$ nix build .#noetic.pluginlib
error: builder for '/nix/store/rqnrly7cjmxylva3fypjn9dq6blbny3r-ros-noetic-class-loader-0.5.0-r1.drv' failed with exit code 1;
       last 25 log lines:
       > -- Using PYTHON_EXECUTABLE: /nix/store/h3i0acpmr8mrjx07519xxmidv8mpax4y-python3-3.12.5/bin/python
       > -- Using default Python package layout
       > -- Found PY_em: /nix/store/293x651b1sv2w9nvi8mxh5av2rcyp18j-python3.12-empy-4.1/lib/python3.12/site-packages/em.py
       > -- Using empy: /nix/store/293x651b1sv2w9nvi8mxh5av2rcyp18j-python3.12-empy-4.1/lib/python3.12/site-packages/em.py
       > -- Using CATKIN_ENABLE_TESTING: OFF
       > -- catkin 0.8.10
       > -- BUILD_SHARED_LIBS is on
       > -- Could NOT find cmake_modules (missing: cmake_modules_DIR)
       > -- Could not find the required component 'cmake_modules'. The following CMake error indicates that you either need to install the package with the same name or change your environment so that it can be found.
       > CMake Error at /nix/store/d6y7v13kik3kl2a673qxpkw0hsm56zxs-ros-noetic-catkin-0.8.10-r1/share/catkin/cmake/catkinConfig.cmake:83 (find_package):
       >   Could not find a package configuration file provided by "cmake_modules"
       >   with any of the following names:
       >
       >     cmake_modulesConfig.cmake
       >     cmake_modules-config.cmake
       >
       >   Add the installation prefix of "cmake_modules" to CMAKE_PREFIX_PATH or set
       >   "cmake_modules_DIR" to a directory containing one of the above files.  If
       >   "cmake_modules" provides a separate development package or SDK, be sure it
       >   has been installed.
       > Call Stack (most recent call first):
       >   CMakeLists.txt:22 (find_package)
       >
       > 
```

--- a/distros/catkin-nixpkgs-prefix-path.patch
+++ b/distros/catkin-nixpkgs-prefix-path.patch
@@ -1,0 +1,11 @@
+--- a/cmake/all.cmake
++++ b/cmake/all.cmake
+@@ -48,6 +48,8 @@ if(NOT DEFINED CMAKE_PREFIX_PATH)
+     else()
+       set(CMAKE_PREFIX_PATH $ENV{CMAKE_PREFIX_PATH})
+     endif()
++  else()
++    string(REPLACE ":" ";" CMAKE_PREFIX_PATH $ENV{NIXPKGS_CMAKE_PREFIX_PATH})
+   endif()
+ endif()
+ message(STATUS "Using CMAKE_PREFIX_PATH: ${CMAKE_PREFIX_PATH}")

--- a/distros/ros1-overlay.nix
+++ b/distros/ros1-overlay.nix
@@ -44,6 +44,13 @@ rosSelf: rosSuper: with rosSelf.lib; {
         url = "https://github.com/ros/catkin/commit/e082348c4992e1850ba5e2bd02bbd7bd0c4c4b82.patch";
         hash = "sha256-NNdV30gNWBf7p8IjyCmnvz9MnU4zFkd4aaXNjs411MA=";
       })
+
+      # Catkin uses CMAKE_PREFIX_PATH to find additional workspaces, but
+      # nixpkgs switched to using NIXPKGS_CMAKE_PREFIX_PATH.
+      # Force CMAKE_PREFIX_PATH to equal NIXPKGS_CMAKE_PREFIX_PATH here.
+      # (https://github.com/NixOS/nixpkgs/commit/8c9c8ade2f88a85ccdd4858cc802d7b7d6c48fe0).
+      # see: https://github.com/lopsided98/nix-ros-overlay/issues/491
+      ./catkin-nixpkgs-prefix-path.patch
     ];
 
     postPatch = postPatch + ''

